lib/cisco_node_utils/client/utils.rb

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env ruby
+#
+# January 2016, Glenn F. Matthews
+#
+# Copyright (c) 2015-2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'client_errors'
+require_relative '../constants'
+require_relative '../logger'
+
+# Utility methods for clients of various RPC formats
+class Cisco::Client
+  # Make a best effort to convert a given input value to an Array.
+  # Strings are split by newlines, and nil becomes an empty Array.
+  def self.munge_to_array(val)
+    val = [] if val.nil?
+    val = val.split("\n") if val.is_a?(String)
+    val
+  end
+
+  def munge_to_array(val)
+    self.class.munge_to_array(val)
+  end
+
+  # Helper function that subclasses may use with get(data_format: :cli)
+  # Method for working with hierarchical show command output such as
+  # "show running-config". Searches the given multi-line string
+  # for all matches to the given value query. If context is provided,
+  # the matches will be filtered to only those that are located "under"
+  # the given context sequence (as determined by indentation).
```

Comment:
Looks like `find-subconfig` could have some unexpected behavior if a context regexp does not match at the left-most column (of the currently trimmed sub-config body text).  Perhaps the documentation for the `get-context` values in the YAML files should say that matching at the beginning of a line (either by including the `^` regexp anchor or by letting the new implicit regexp behavior do it for you) is *required*, not just a best practice?